src/uri/UriValidator.cpp

```diff
@@ -0,0 +1,159 @@
+    status.set_code(UCode::INVALID_ARGUMENT);
+    status.set_message("Uri is missing uSoftware Entity name.");
+    return status;
+  }
+  status.set_code(UCode::OK);
+  status.set_message("validate Success.");
+  return status;
+}
+
+/**
+ * Validate a {@link UUri} that is meant to be used as an RPC method URI. Used in Request sink values and Response source values.
+ *
+ * @param uri {@link UUri} to validate.
+ * @return Returns Status containing a success or a failure with the error message.
+ */
+UStatus UriValidator::validateRpcMethod(const UUri& uri) {
+    UStatus status = UriValidator::validate(uri);
+    if (status.code() != UCode::OK) {
+        return status;
+    }
+    if (!UriValidator::isRpcMethod(uri)) {
+      status.set_code(UCode::INVALID_ARGUMENT);
+      status.set_message("Invalid RPC method uri. Uri should be the method to be called, or method from response.");
+      return status;
+    }
+  return status;
+}
+
+/**
+ * Validate a {@link UUri} that is meant to be used as an RPC method URI. Used in Request sink values and Response source values.
+ *
+ * @param uri {@link UUri} to validate.
+ * @return Returns Status containing a success or a failure with the error message.
+ */
+UStatus UriValidator::validateRpcResponse(const UUri& uri) {
+    UStatus status = UriValidator::validate(uri);
+    if (status.code() != UCode::OK) {
+        return status;
+    }
+    if (!UriValidator::isRpcResponse(uri)) {
+        status.set_code(UCode::INVALID_ARGUMENT);
+        status.set_message("Invalid RPC response type.");
+        return status;
+    }
+    
+    return status;
+}
+
+/**
+ * Returns true if URI is of type RPC.
+ *
+ * @param uri {@link UUri} to check if it is of type RPC method
+ * @return Returns true if URI is of type RPC.
+ */
+
+bool UriValidator::isRpcMethod(const UUri& uri) {
+    return (nullptr != &uri && 
+     !isEmpty(uri.resource()) &&
+     uri.resource().has_instance() && 
+     !uri.resource().instance().empty()) ||
+     (uri.resource().has_id() && 
+     uri.resource().id() != 0 && 
+     uri.resource().id() < 1000);
+}
+
+/**
+ * Returns true if URI contains both names and numeric representations of the names inside its belly.
+ * Meaning that this UUri can be serialized to long or micro formats.
+ *
+ * @param uri {@link UUri} to check if resolved.
+ * @return Returns true if URI contains both names and numeric representations of the names inside its belly.
+ * Meaning that this UUri can buree serialized to long or micro formats.
+ */
+bool UriValidator::isResolved(const UUri& uri) {
+  return !isEmpty(uri) && isLongForm(uri) && isMicroForm(uri);
+}
+
+/**
+ * Returns true if URI is of type RPC response.
+ *
+ * @param uri {@link UUri} to check response
+ * @return Returns true if URI is of type RPC response.
+ */
+bool UriValidator::isRpcResponse(const UUri& uri) {
+  if (nullptr == &uri)
+  {
+    return false;
+  }
+  return (uri.resource().name().find("rpc") &&
```

Comment:
Same comment on return in isRpcMethod() above.

Reply:
Done